packages/beacon-node/src/chain/produceBlock/validateBlobsAndKzgCommitments.ts

```diff
@@ -1,16 +1,54 @@
-import {ExecutionPayload} from "@lodestar/types";
+import {CELLS_PER_EXT_BLOB, ForkPostDeneb, ForkSeq} from "@lodestar/params";
+import {ExecutionPayload, fulu} from "@lodestar/types";
 import {BlobsBundle} from "../../execution/index.js";
+import {ckzg} from "../../util/kzg.js";
 
 /**
  * Optionally sanity-check that the KZG commitments match the versioned hashes in the transactions
  * https://github.com/ethereum/consensus-specs/blob/11a037fd9227e29ee809c9397b09f8cc3383a8c0/specs/eip4844/validator.md#blob-kzg-commitments
  */
 
-export function validateBlobsAndKzgCommitments(_payload: ExecutionPayload, blobsBundle: BlobsBundle): void {
-  // sanity-check that the KZG commitments match the blobs (as produced by the execution engine)
+export function validateBlobsAndKzgCommitments(
+  fork: ForkPostDeneb,
+  _payload: ExecutionPayload,
+  blobsBundle: BlobsBundle,
+  cells?: fulu.Cell[][]
+): void {
   if (blobsBundle.blobs.length !== blobsBundle.commitments.length) {
     throw Error(
       `Blobs bundle blobs len ${blobsBundle.blobs.length} != commitments len ${blobsBundle.commitments.length}`
     );
   }
+
+  if (ForkSeq[fork] < ForkSeq.fulu) {
+    if (blobsBundle.proofs.length !== blobsBundle.blobs.length) {
+      throw new Error(
+        `Invalid proofs length for BlobsBundleV1 format: expected ${blobsBundle.blobs.length}, got ${blobsBundle.proofs.length}`
+      );
+    }
+    return;
```

Comment:
lets add blob batch proof validation here as well since we added for cells

Reply:
Updated with blob batch proof verification